db/madmin_conversion.py

```diff
@@ -0,0 +1,306 @@
+TABLES = [
+    """CREATE TABLE IF NOT EXISTS `madmin_instance` (
+        `instance_id` INT NOT NULL AUTO_INCREMENT,
+        `name` VARCHAR(128) NOT NULL,
+        PRIMARY KEY (`instance_id`),
+        UNIQUE KEY (`name`, `instance_id`)
+    ) ENGINE = InnoDB;""",
+
+    """CREATE TABLE IF NOT EXISTS `settings_monivlist` (
+        `monlist_id` int(11) NOT NULL AUTO_INCREMENT,
+        `guid` varchar(32) NULL,
+        `instance_id` int(11) NOT NULL,
+        `name` varchar(128) NOT NULL,
+        PRIMARY KEY (`monlist_id`),
+        -- UNIQUE KEY (`name`, `instance_id`),
+        CONSTRAINT `fk_mil_instance`
+            FOREIGN KEY (`instance_id`) 
+            REFERENCES `madmin_instance` (`instance_id`)
+            ON DELETE CASCADE
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;""",
+
+    """CREATE TABLE IF NOT EXISTS `settings_monivlist_to_mon` (
+        `monlist_id` int(11) NOT NULL,
+        `mon_id` int(11) NOT NULL,
+        `mon_order` int(11) NOT NULL,
+        PRIMARY KEY (`monlist_id`,`mon_id`),
+        INDEX (`monlist_id`),
+        INDEX (`mon_id`),
+        CONSTRAINT `settings_monivlist_to_mon_ibfk_1`
+            FOREIGN KEY (`monlist_id`)
+            REFERENCES `settings_monivlist` (`monlist_id`)
+            ON DELETE CASCADE
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;""",
+
+    """CREATE TABLE IF NOT EXISTS `settings_auth` (
+        `auth_id` int(11) NOT NULL AUTO_INCREMENT,
+        `guid` varchar(32) NULL,
+        `instance_id` int(11) NOT NULL,
+        `username` varchar(32) COLLATE utf8mb4_unicode_ci NOT NULL,
+        `password` varchar(32) COLLATE utf8mb4_unicode_ci NOT NULL,
+        PRIMARY KEY (`auth_id`),
+        -- UNIQUE (`instance_id`, `username`, `password`),

```

Comment:
This unique key is commented out. 

I would also not do a unique over these 3 fields but just over `instance_id` and `username`.

Reply:
I am on the fence about these unique keys for MADmin settings.  They are the correct way to go as they can cause unintended consequences.  However its how the current config works and to my knowledge no one has complained so far.  I think we need input from @sn0opy or @cecpk to figure out the route they would like us to take.

Reply:
Eventually it wouldn't matter to have multiple accounts with the same name and different passwords. It appears just "uncommon" in general...